kernel/src/time/clocks/system_wide.rs

```diff
@@ -144,47 +144,75 @@ impl BootTimeClock {
 }
 
 impl Clock for JiffiesClock {
-    fn read_time(&self) -> Duration {
-        Jiffies::elapsed().as_duration()
+    fn read_duration(&self) -> Duration {
+        self.read_jiffies().into()
+    }
+
+    fn read_jiffies(&self) -> Jiffies {
+        Jiffies::elapsed()
     }
 }
 
 impl Clock for RealTimeClock {
-    fn read_time(&self) -> Duration {
+    fn read_duration(&self) -> Duration {
         SystemTime::now()
             .duration_since(&SystemTime::UNIX_EPOCH)
             .unwrap()
     }
+
+    fn read_jiffies(&self) -> Jiffies {
+        Jiffies::from_duration(self.read_duration())
+    }
 }
 
 impl Clock for MonotonicClock {
-    fn read_time(&self) -> Duration {
+    fn read_duration(&self) -> Duration {
         read_monotonic_time()
     }
+
+    fn read_jiffies(&self) -> Jiffies {
+        Jiffies::from_duration(self.read_duration())
+    }
 }
 
 impl Clock for RealTimeCoarseClock {
-    fn read_time(&self) -> Duration {
+    fn read_duration(&self) -> Duration {
         *Self::current_ref().get().unwrap().disable_irq().lock()
     }
+
+    fn read_jiffies(&self) -> Jiffies {
+        Jiffies::from_duration(self.read_duration())
+    }
 }
 
 impl Clock for MonotonicCoarseClock {
-    fn read_time(&self) -> Duration {
-        RealTimeCoarseClock::get().read_time() - *START_TIME_AS_DURATION.get().unwrap()
+    fn read_duration(&self) -> Duration {
+        RealTimeCoarseClock::get().read_duration() - *START_TIME_AS_DURATION.get().unwrap()
+    }
+
+    fn read_jiffies(&self) -> Jiffies {
+        Jiffies::from_duration(self.read_duration())
     }
 }
 
 impl Clock for MonotonicRawClock {
-    fn read_time(&self) -> Duration {
+    fn read_duration(&self) -> Duration {
         read_monotonic_time()
     }
+
+    fn read_jiffies(&self) -> Jiffies {
+        Jiffies::from_duration(self.read_duration())
```

Comment:
It appears that modifying the `Clock` trait affects clocks other than `CpuClock` and `ProfClock`. This is not intended and seems unnecessary. If so, it would be better to follow the previous approach to implement `read_jiffies` only for the appropriate clock types (but of course, we should still use `add_jiffies` instead of `add_time` to avoid unnecessary conversion and information loss).

Sorry for the noise caused by my previous comment. I should have checked all `impl`s of the `Clock` trait before writing that.